index.js

```diff
@@ -2312,11 +2312,41 @@ function measureText(font, text) {
 
 /**
  * Writes the image to a file
- * @param path a path to the destination file (either PNG or JPEG)
- * @param (optional) cb a function to call when the image is saved to disk
- * @returns this for chaining of methods
+ *
+ * @param {string} path a path to the destination file (either PNG or JPEG)
+ * @param {null|function} cb a function to call when the image is saved to disk
+ * @param {boolean} returnPromise
+ * @returns {Jimp|Promise} if returnPromise is true then returns a Promise otherwise `this` for chaining methods
+ */
+Jimp.prototype.write = function (path, cb, returnPromise) {
```

Comment:
I think it makes sense to have a promise implementation for ```write()```, but I agree backwards compatibility is an issue, and it's strange to have to call Jimp with an empty callback when using the promise implementation.

In the case of Jimp, I believe this also breaks functionality when you want multiple outputs. Though this isn't how I've used Jimp myself, can't you do this for multiple thumbnail processing, and expect it to work?

```
.resize(1000,1000)
.write("/path/to/1000px-thumbnail")
.resize(500,500)
.write("/path/to/500px-thumbnail")
```

... at the end of the day, promises would be great, but the implementation here for ```.write()``` is problematic.